main.go

```diff
@@ -123,22 +86,45 @@ func runTemplate(c *cli.Context) error {
 
 	g := NewGomplate(data, lDelim, rDelim)
 
-	inputs := readInputs(c.String("in"), c.StringSlice("file"))
+	if err := validateInOutOptions(c); err != nil {
+		return err
+	}
 
-	outputs := c.StringSlice("out")
-	if len(outputs) == 0 {
-		outputs = []string{"-"}
+	if c.String("input-dir") != "" {
+		return processInputDir(c, g)
 	}
 
-	for n, input := range inputs {
-		out, err := openOutFile(outputs[n])
-		if err != nil {
-			return err
-		}
-		defer out.Close() // nolint: errcheck
-		g.RunTemplate(input, out)
+	return processInputFiles(c, g)
+}
+
+// Called from process ...
+func renderTemplate(g *Gomplate, inString string, outPath string) error {
+	outFile, err := openOutFile(outPath)
+	if err != nil {
+		return err
 	}
+	defer checkClose(outFile, &err)
+	g.RunTemplate(inString, outFile)
+	return nil
+}
 
+func validateInOutOptions(c *cli.Context) error {
```

Comment:
I _think_ I might need to take this back. I started working on it in #122 but it seems to have trouble with taking `-` as an argument for `--file` or `--out`. I think it maybe makes sense to keep this code in for now and not block this PR on the other one.